### PR TITLE
Add Bazel Build

### PR DESCRIPTION
**Pull request by ahirreddy**
Changes:
- Basic .gitignore file
- Gazelle generated BUILD and WORKSPACE
- Update script that automates all the gazelle commands

### CODE REVIEW

File: update_gazelle.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+bazel run //:gazelle -- update -go_prefix "github.com/bazel-cache-s3"
+bazel run //:gazelle -- update-repos github.com/aws/aws-sdk-go
+bazel run //:gazelle -- update-repos github.com/go-chi/chi
+bazel run //:gazelle -- update-repos github.com/golang/groupcache
+bazel run //:gazelle -- update-repos github.com/pkg/errors
+bazel run //:gazelle -- update-repos gopkg.in/tylerb/graceful.v1
```
Comment (ahirreddy) [inline update_gazelle.sh:9]: Can't collapse the update-repos calls into one command. There's some gazelle issue that has it write out the first repo N times.

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,47 @@
+load("@bazel_gazelle//:def.bzl", "gazelle")
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+gazelle(
+    name = "gazelle",
+    prefix = "github.com/bazel-cache-s3",
```
Comment (thomasdesr) [inline BUILD]: This should probably be `github.com/thomaso-mirodin/bazel-cache-s3` so as to mirror the actual import path

**Review by thomasdesr — CHANGES_REQUESTED** [review]
Quick fix on the import path otherwise lgtm!

File: BUILD (hunk 2 above)
Comment (ahirreddy) [reply BUILD]: Fixed

**Pull request merged**